### PR TITLE
Add form submission backend [WIP]

### PR DESCRIPTION
**Pull request by dinoocch**
This commit adds the ability to submit a form via the web interace and
parse the resulting json data into a model.  In addition, we rely on the
model validation to check the forms correctness.

- [X]  Form Submission
- [ ]  Redirection
- [ ]  Error Handling

### CODE REVIEW

**Comment by dinoocch** [comment]
@m-wynn Are we using the same snack bar style for error messages as in Orderboard?

**Comment by m-wynn** [comment]
@dinoocch sure, that's fine, unless you have a better idea.

**Comment by dinoocch** [comment]
On form submit, the view will redirect the user to either:
- [ ] a list of forms
- [ ] the form submission page for the created form
- [x] the rendered form preview
* Not sure which is better...

**Comment by the-maldridge** [comment]
I'm inclined to redirect to a page that shows the rendered version of what was just entered.  Probably the same page we'd use for printing since then we can kill two birds with one stone.

File: constellation_forms/views.py [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
+        groups = [(g.name, g.pk) for g in Group.objects.all()]
+
+        return render(request, 'constellation_forms/create-form.html', {
+            'groups': groups,
+            'template_settings': template_settings,
+        })
 
+    def post(self, request):
+        ''' Creates a form '''
+        form_data = json.dumps(request.body)
```
Comment (m-wynn) [inline constellation_forms/views.py]: Shouldn't this be `form_data = json.loads(request.POST['data'])`?

**Comment by m-wynn** [comment]
json.dumps still needs to be json.loads

**Comment by dinoocch** [comment]
Woops, you're correct, I will fix when I get back... >.<

**Pull request closed without merging**